File: glTest/src/glTest/problems/NullProblem.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package glTest.problems;

import static com.jogamp.opengl.GL2ES3.GL_COLOR;
import static com.jogamp.opengl.GL2ES3.GL_DEPTH;
import com.jogamp.opengl.GL4;
import glTest.framework.ApplicationState;

/**
 *
 * @author GBarbieri
 */
public class NullProblem extends Problem {

    @Override
    public boolean init(GL4 gl4) {

        super.init(gl4);

        clearColor.put(0, 0.2f).put(1, 0.0f).put(2, 0.0f).put(3, 1.0f);
        clearDepth.put(0, 1.0f);

        return true;
    }

    @Override
    public void render(GL4 gl4) {

        gl4.glClearBufferfv(GL_COLOR, 0, clearColor);
        gl4.glClearBufferfv(GL_DEPTH, 0, clearDepth);

        // Nothing, because it's the NULL solution.
    }

    @Override
    public String getName() {
        return "NullProblem";
    }

}

File: src/main/kotlin/main/apiTest/problems/NullProblem.kt
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package main.apiTest.problems

import glm_.vec4.Vec4
import main.apiTest.solutions.NullSolution
import main.apiTest.solutions.Solution

/**
 *
 * @author GBarbieri
 */
class NullProblem : Problem() {

    override val name get() = "NullProblem"

    override fun init() = true  // Nothing to initialize

    override fun render() {} // Nothing, because it's the NULL solution.

    override fun getClearValues(outCol: Vec4): Float {
        outCol.put(0.2f, 0f, 0f, 1f)
        return 1f // outDepth
    }

    override fun setSolution(solution: Solution?): Boolean {
        if (!super.setSolution(solution))
            return false

        activeSolution?.let { sol ->
            println("Solution ${sol.name} - Initializing.")
            return (sol as NullSolution).init().also { ret ->
                println("Solution ${sol.name} - Initialize complete (Success: $ret).")
            }
        }

        return true
    }
}
